prometheus_client/core.py

```diff
@@ -198,6 +198,11 @@ def add_metric(self, labels, value):
           labels: A list of label values
           value: The value of the metric.
         '''
+        try:
```

Comment:
It'd be more Pythonic to duck type and to convert a string, rather than limit to two particular classes.

This needs a unittest